### DISCUSSION TITLE
Uno Community Toolkit WinUI

### DISCUSSION THREAD

**Discussion by esobrino** [unoplatform/uno#8810] [Q&A] [answered]
Created an Uno WinUI project and...

I added the "Uno.CommunityToolkit.WinUI.UI.Controls" in the "Wasm" project and tests successfully, it is coded as follows:

<UserControl
   x:Class="MyProj.UserControl1"
   xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
   xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
   xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
   xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
   xmlns:tk="using:Uno.CommunityToolkit.WinUI.UI.Controls"
   mc:Ignorable="d">

   <tk:DataGrid>
      <tk:DataGrid.Columns>
         <tk:DataGridTextColumn Header="Rank" />
         <tk:DataGridComboBoxColumn Header="Mountain" />
      </tk:DataGrid.Columns>
   </tk:DataGrid>
</UserControl>

The above is GOOD.  

Now, I am trying the same thing in the "Windows" project but I keep getting "... CommunityToolkit' could not be found in the global namespace...".  I included the same library ("Uno.CommunityToolkit.WinUI.UI.Controls").  

Can someone provide some clues?

**Comment by jeromelaban** [comment, marked as answer]
Please provide a repro sample of what you're trying. Thanks!

**Reply by esobrino** [reply]
The test I was doing is not much.  For now, I am concentrating in 2 targets "Wasm" and "Windows" only.

I performed the following steps:

1. Create an Uno WinUI project following steps in "dotnet new unoapp-winui-net6 -o MyUnoApp" as detailed in "https://platform.uno/docs/articles/get-started-winui3.html".
2. Added the "Uno.CommunityToolkit.WinUI.UI.Controls" library to both "Wasm" and "Windows" projects.
3. Added a User-Control in the Shared area as shown in the provided UserControl1.txt file.
4. Compiled and test in "Wasm"... all good.
5. Compiled in "Windows" and got "Windows" project but I keep getting "... CommunityToolkit' could not be found in the global namespace..." error so I couldn't test it.

Thats it.


.
[UserControl1.txt](https://github.com/unoplatform/uno/files/8716351/UserControl1.txt)

**Reply by jeromelaban** [reply]
Thanks. "Not much" is generally the important part where details matter. Make sure to follow the directions here: https://platform.uno/docs/articles/uno-community-toolkit.html?tabs=tabid-vswin

Those directions apply to the UWP heads, but you can do the same with the WinUI heads.

**Reply by esobrino** [reply]
The quick response is greatly appreciated!

I did use that reference for my example.  I finally got it to work properly... my confusion was related to how to reference the assembly in XAML, but following the instructions you just use "using:CommunityToolKit.WinUI.UI.Controls" for both "Wasm" and "Windows".

All working now!

**Reply by ApoorvaPetkar** [reply]
Thank you, worked for me
